core/lib/lh-error.js

```diff
@@ -107,14 +107,14 @@ const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
 const LHERROR_SENTINEL = '__LighthouseErrorSentinel';
 const ERROR_SENTINEL = '__ErrorSentinel';
 /**
- * @typedef {{sentinel: '__LighthouseErrorSentinel', code: string, stack?: string, [p: string]: string|undefined}} SerializedLighthouseError
+ * @typedef {{sentinel: '__LighthouseErrorSentinel', code: string, stack?: string, [p: string]: unknown}} SerializedLighthouseError
  * @typedef {{sentinel: '__ErrorSentinel', message: string, code?: string, stack?: string}} SerializedBaseError
  */
 
 class LighthouseError extends Error {
   /**
    * @param {LighthouseErrorDefinition} errorDefinition
-   * @param {Record<string, string|undefined>=} properties
+   * @param {Record<string, unknown>=} properties
```

Comment:
Why is this change necessary?

Reply:
initially I needed this change when the base tsconfig target was updated to 2022, reverted in https://github.com/GoogleChrome/lighthouse/pull/14911/commits/de6a3944ae73a6b27b12e23059728461f214b642 . the base tsconfig didn't need the change anymore after I moved the truncate function to `shared/util.js` (it _was_ in `page-functions.js`). Might as well keep this more correct type, though.